net-libs/libtorrent-rasterbar: add 2.0.13 with improvements, add myself as proxied maintainer

fkobi opened the pull request:
This introduces `USE=+deprecated`.
My final goal is to disable it by default so that the deprecated stuff gets built only when it needs to.

EDIT: this PR does not do that but I'd like to add that

---

Please check all the boxes that apply:

- [x] I can submit this contribution in agreement with the [Copyright Policy](https://www.gentoo.org/glep/glep-0076.html#certificate-of-origin).
- [x] I have certified the above via adding a `Signed-off-by` line to *every* commit in the pull request.
- [x] This contribution has not been created with the assistance of Natural Language Processing artificial intelligence tools, in accordance with the [AI policy](https://wiki.gentoo.org/wiki/Project:Council/AI_policy).
- [x] I have run `pkgcheck scan --commits --net` to check for issues with my commits.

Please note that all boxes must be checked for the pull request to be merged.

fkobi committed:
net-libs/libtorrent-rasterbar: improve metadata

- add:
  - longdescription
  - upstream maintainer
  - bugs-to, changelog, doc
- improve USE="dht examples" descriptions
- delete old remote-id

Signed-off-by: Filip Kobierski <fkobi@pm.me>

fkobi committed:
net-libs/libtorrent-rasterbar: add myself as proxied maintainer

Signed-off-by: Filip Kobierski <fkobi@pm.me>

thesamesam commented:
I think Eli may be best-placed to review but also might make sense for him to co-maintain if he's willing given qbittorrent is the primary consumer.

fkobi committed:
net-libs/libtorrent-rasterbar: add 2.0.13

- sort mycmakeargs
- fix LICENSE
- remove einstalldocs -- cmake_src_install does that already
- add DOCS
- require patchelf on USE=examples
- style: unfold
- remove boost-python-module-name as it is no longer needed
- remove BUILD_SHARED_LIBS as it's the default
- remove explicit C++ standard assignment as it just requires >=C++14

Closes: https://bugs.gentoo.org/969058
Signed-off-by: Filip Kobierski <fkobi@pm.me>
